docs/en/engines/database-engines/overlay.md

```diff
@@ -0,0 +1,167 @@
+
+---
+
+## Construction {#construction}
+
+### Factory (SQL) {#sql}
+
+```sql
+CREATE DATABASE dboverlay ENGINE = Overlay('db_overlay_a', 'db_overlay_b');
+```
+
+* The factory resolves every underlying DB name via `DatabaseCatalog::tryGetDatabase(name)`.
+* **Null checks are enforced**: a missing underlying DB throws `BAD_ARGUMENTS` with a clear message.
+* Self-reference is rejected.
+
+---
+
+## UUID semantics {#uuid}
+
+* **Facade Over Catalog**: `getUUID()` returns **`UUIDHelpers::Nil`**.
+  This makes the overlay **skip DB-UUID registration** in the global catalog, eliminating collisions with DB/table UUIDs.
+* **Owned Members**: `getUUID()` returns the **first member’s non-Nil UUID**, preserving the behavior expected by `clickhouse-local`.
+
+---
+
+## What operations does the overlay support? {#implementation}
+
+### Table/DB discovery {#discovery}
+
+* `SHOW TABLES FROM dboverlay` — returns the **union** of member DB tables.
+* `SELECT … FROM dboverlay.table` — **reads** transparently hit the underlying table.
+
+### Mutating operations (facade mode) {#operations}
+
+| Operation                  | Behavior                                                                                  |
+| :------------------------- | :---------------------------------------------------------------------------------------- |
+| `CREATE TABLE dboverlay.*` | **Rejected** — throws `BAD_ARGUMENTS` with a clear message to create in an underlying DB. |
+| `ATTACH TABLE dboverlay.*` | **Rejected** — `BAD_ARGUMENTS`.                                                           |
+| `ALTER TABLE dboverlay.*`  | **Pass-through** — forwards to the appropriate underlying DB.                             |
+| `RENAME TABLE dboverlay.*` | **Rejected** — `BAD_ARGUMENTS`.                                                           |
+| `DROP TABLE dboverlay.*`   | **No-op** — ignored (allows `DROP DATABASE dboverlay` to succeed).                        |
+| `INSERT INTO dboverlay.*`  | **Pass-through** — executes against the table in the corresponding underlying DB.         |
+
+> Rationale: the facade is a **view**. Data-definition & data-mutation happen in the member databases.
+
+### Database life cycle (facade mode) {#life-cycle}
+
+* `DROP DATABASE dboverlay` — **succeeds** and **does not** cascade to members.
+
+  * Implementation details that make this safe:
+
+    * `getUUID()` returns **Nil** (no DB UUID mapping).
+    * `getTablesIterator(ctx, filter, /*skip_not_loaded=*/true)` returns **empty** (so the dropper doesn’t see tables).
+    * `empty()` returns **true** (facade reports empty during destructive passes).
+    * `dropTable()` is a **no-op** (dropper won’t fail mid-way).
+    * `drop()` is a **no-op** (doesn’t touch member DBs).
+* `isReadOnly()` returns **false** (so the server doesn’t block `DROP DATABASE` up front). Mutating table ops are still guarded individually.
```

Comment:
Hm, but you return `readonly`, so it is dependent

Reply:
Updated to mention that